x-pack/plugins/rule_registry/server/utils/create_lifecycle_executor.ts

```diff
@@ -258,6 +260,15 @@ export const createLifecycleExecutor = (logger: Logger, ruleDataClient: RuleData
     event[ALERT_START] = started;
     event[ALERT_UUID] = alertUuid;
 
+    // not sure why typescript needs the non-null assertion here
+    // we already assert the value is not undefined with the ternary
+    // still getting an error with the ternary.. strange.
+
+    event[SPACE_IDS] =
+      event[SPACE_IDS] == null
+        ? [spaceId]
+        : [spaceId, ...event[SPACE_IDS]!.filter((sid) => sid !== spaceId)];
```

Comment:
It looks like we're only adding space IDs for lifecycle alerts. Is there something preventing us from injecting the space ID inside the RuleDataClient `bulk` method so that persistence rule types and any future rule types will also automatically inject the space ID?

Reply:
This is true we could push down the space id from the plugins into the rule data client, which would then be accessible via the bulk method. I'll take a look at this.

Reply:
We would end up iterative over the data passed into bulk twice rather than once, so not that optimal.